scaffold/bitbucket/bitbucket-pipelines.example.yml

```diff
@@ -0,0 +1,156 @@
+                        }
+                      ]
+                    }'
+                # Create "Report Annotations" for each change in composer.lock.
+                - |
+                  while read ROW; do
+                    case "$ROW" in
+                      +* | ''                ) ;; # Skip lines without data.
+                      '| Production Changes'*) IS_DEV_DEPENDENCY=0 ;;
+                      '| Dev Changes'*       ) IS_DEV_DEPENDENCY=1 ;;
+                      *                      ) # Process rows with package info.
+                        # Remove spaces.
+                        printf -v ROW '%s' $ROW
+                        # Explode row into array of columns.
+                        IFS='|' read -ra COLS <<< "$ROW"
+                        if [ "${COLS[2]}" = 'NEW' ] ; then
+                          ANNOTATION="${COLS[1]} added @ v${COLS[3]}"
+                        else
+                          ANNOTATION="${COLS[1]} changed: v${COLS[2]} ⇾ v${COLS[3]}"
+                        fi
+                        if [ $IS_DEV_DEPENDENCY -eq 1 ] ; then
+                          ANNOTATION="$ANNOTATION (dev dependency)"
+                        fi
+                        ANNOTATION_ID="${REPORT_ID}-$(echo "${COLS[1]}" | sed 's/\///g')"
+                        curl \
+                          --proxy 'http://localhost:29418' \
+                          --request PUT \
+                          --url "http://api.bitbucket.org/2.0/repositories/$BITBUCKET_REPO_OWNER/$BITBUCKET_REPO_SLUG/commit/$BITBUCKET_COMMIT/reports/$REPORT_ID/annotations/$ANNOTATION_ID" \
+                          --header 'Accept: application/json' \
+                          --header 'Content-Type: application/json' \
+                          --data-raw "$( jq -n \
+                            --arg title "composer-lock-diff" \
+                            --arg annotation_type "CODE_SMELL" \
+                            --arg summary "$ANNOTATION" \
+                            --arg path "composer.lock" \
+                            '{
+                              title: $title,
+                              annotation_type: $annotation_type,
+                              summary: $summary,
+                              path: $path
+                            }' \
+                          )"
+                        ;;
+                    esac
+                  done <<< "$COMPOSER_LOCK_DIFF"
+      # The following parallel group will only start after all above steps
+      # have finished successfully.
+      - parallel:
+          - step:
+              name: 'Functional tests'
+              caches:
+                - composer
+              script:
+                - ./vendor/bin/task test:functional
+          - step:
+              name: 'Check Lighthouse reports'
+              script:
+                # Install the Tugboat CLI tool.
+                - wget https://dashboard.tugboat.qa/cli/linux/tugboat.tar.gz
```

Comment:
wut. This is neat! I had no idea we could get the lighthouse data via Tugboat's API.